docs/doxygen/demos/pkcs11_demo.dox

```diff
@@ -0,0 +1,41 @@
+/**
+@page pkcs11_demo corePKCS11
+@brief Demos used to illustrate various functionalities of the corePKCS11 library.
+
+@section pkcs11_demo_rng PKCS #11 Random Number Generation Demo
+@brief Demo of using the PKCS #11 interface to generate a random number.
+
+
+This demo will introduce the basics of PKCS #11. It will step you through the data types as well as giving
```

Comment:
```suggestion
This demo will introduce the basics of PKCS #11. It will step you through the data types as well as give
```
"give" sounds better to me lol